core/src/data/value/mod.rs

```diff
@@ -635,11 +635,17 @@ impl Value {
         }
     }
 
-    pub fn like(&self, other: &Value, case_sensitive: bool) -> Result<Value> {
+    pub fn like(&self, other: &Value, case_sensitive: bool, escape_char: char) -> Result<Value> {
         use Value::*;
 
         match (self, other) {
-            (Str(a), Str(b)) => a.like(b, case_sensitive).map(Bool),
+            (Str(a), Str(b)) => {
+                if escape_char == '%' || escape_char == '#' {
+                    a.like_with_escape(b, case_sensitive, escape_char).map(Bool)
+                } else {
+                    a.like(b, case_sensitive).map(Bool)
+                }
```

Comment:
It would be better to handle escape_char in only one place. (now it becomes two: `value/mod.rs` and `data/string_ext.rs`.
currently that is handled in `data/string_ext.rs` so we'd better to make `StringExt::like` to always take `escape_char`. rather than adding a new function `StringExpr::like_with_escape`.